WCFDataService/Client/System/Data/Services/Client/MaterializeFromAtom.cs

```diff
@@ -69,6 +69,11 @@ internal class MaterializeAtom : IDisposable, IEnumerable, IEnumerator
         private bool moved;
 
         /// <summary>
+        /// The ODataResponseMessage to dispose when we are done.
+        /// </summary>
+        private IODataResponseMessage responseMessage;
```

Comment:
Seems there is a side effect of the fix is that when we are doing some operation, like create an entity, it will return 201, and we will not do enumeration after the operation, then the message will not dispose.